packages/query-core/src/utils.ts

```diff
@@ -246,6 +246,15 @@ export function replaceEqualDeep(a: any, b: any): any {
     return a
   }
 
+  if (process.env.NODE_ENV !== 'production') {
+    try {
+      JSON.stringify(a)
+      JSON.stringify(b)
+    } catch {
+      throw new Error('Values are not serializable')
```

Comment:
let's please add a good `console.error` log here, like we do in other places as well. Otherwise, it's up to the user to inspect this error and it might get unnoticed.

this likely means that the test will need to stub the console and assert that the log happened. We have some good examples of this already

Reply:
@TkDodo, Okay. I just thought that the idea of this catch was an early throw of error instead of waiting until the call stack is exceeded and the query transitions to an error state.
So, it's rather just a hint for the developer in this case, right?

Reply:
I definitely want the error log. I don't really care if we throw or not 🤷‍♂️  (in dev mode).

Reply:
@TkDodo ok, I've added the `console.error`, but also left the `throw` to initiate the early transition to `error` status. Also, I lifted this check a bit to access the query `options`.